drivers/sensor/nrf5/temp_nrf5.c

```diff
@@ -25,40 +25,50 @@ LOG_MODULE_REGISTER(temp_nrf5, CONFIG_SENSOR_LOG_LEVEL);
 struct temp_nrf5_data {
 	struct k_sem device_sync_sem;
 	int32_t sample;
-	struct device *clk_dev;
+	struct onoff_service *clk_srv;
 };
 
-static void hfclk_on_callback(struct device *dev, clock_control_subsys_t subsys,
-			      void *user_data)
+static void hfclk_on_callback(struct onoff_service *srv,
+			      struct onoff_client *cli,
+			      void *user_data,
+			      int res)
 {
-	nrf_temp_task_trigger(NRF_TEMP, NRF_TEMP_TASK_START);
+	bool *do_start = user_data;
+
+	/* Callback will be called twice: on start and stop. Use flag to ensure
+	 * that sensor is started only on start.
+	 */
+	if (*do_start) {
+		nrf_temp_task_trigger(NRF_TEMP, NRF_TEMP_TASK_START);
+		*do_start = false;
+	}
 }
 
 static int temp_nrf5_sample_fetch(struct device *dev, enum sensor_channel chan)
 {
 	struct temp_nrf5_data *data = dev->driver_data;
-	struct clock_control_async_data clk_data = {
-		.cb = hfclk_on_callback
-	};
+	struct onoff_client cli;
+	bool do_start = true;
 
 	int r;
 
 	/* Error if before sensor initialized */
-	if (data->clk_dev == NULL) {
+	if (data->clk_srv == NULL) {
 		return -EAGAIN;
 	}
 
 	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_DIE_TEMP) {
 		return -ENOTSUP;
 	}
 
-	r = clock_control_async_on(data->clk_dev, CLOCK_CONTROL_NRF_SUBSYS_HF,
-					&clk_data);
-	__ASSERT_NO_MSG(!r);
+	onoff_client_init_callback(&cli, hfclk_on_callback, &do_start);
```

Comment:
Changes to `temp_nrf5.c` should be extracted from this commit and merged with those in commit 3b13686324e1a6a06041294453517bca8a838985 (drivers: sensor: nrf5: Adapt temp driver to use onoff for clock), because currently they use no longer existing onoff API.